google-cloud-storage/src/test/java/com/google/cloud/storage/it/runner/registry/StorageInstance.java

```diff
@@ -48,45 +47,10 @@ public Object get() {
     return proxy;
   }
 
-  private static final Object[] updateParameters = {BucketInfo.class, BucketTargetOption[].class};
-  private static final Object[] deleteParameters = {String.class, BucketSourceOption[].class};
-
   @Override
   public void start() {
     storage = options.getService();
-    // Define a proxy which can veto calls which attempt to mutate protected buckets
-    //   this helps guard against a test trying to mutate the global bucket rather than creating its
-    //   own bucket.
-    proxy =
-        Reflection.newProxy(
-            Storage.class,
-            ((proxy1, method, args) -> {
-              String methodName = method.getName();
-              Class<?>[] parameterTypes = method.getParameterTypes();
-              if (methodName.equals("update")
-                  && Arrays.deepEquals(parameterTypes, updateParameters)) {
-                BucketInfo bucketInfo = (BucketInfo) args[0];
-                if (protectedBucketNames.isProtected(bucketInfo.getName())) {
-                  throw err(bucketInfo.getName());
-                }
-              } else if (methodName.equals("delete")
-                  && Arrays.deepEquals(parameterTypes, deleteParameters)) {
-                String bucketName = (String) args[0];
-                if (protectedBucketNames.isProtected(bucketName)) {
-                  throw err(bucketName);
-                }
-              } else if (methodName.equals("setIamPolicy")) {
-                String bucketName = (String) args[0];
-                if (protectedBucketNames.isProtected(bucketName)) {
-                  throw err(bucketName);
-                }
-              }
-              try {
-                return method.invoke(storage, args);
-              } catch (InvocationTargetException e) {
-                throw e.getCause();
-              }
-            }));
+    proxy = new VetoingStorageProxy();
```

Comment:
Side bar: 
I'm wondering if you're planning on exposing AbstractStorageProxy as a builder option that can be passed and potentially used for other efforts (thinking observability)?

Reply:
At this point, no. I think observability stuff we're going to want/need to be more explicit and prescriptive about any extension points we provide. That said, for our internal implementation we could so something similar to allow for a single implementation that would wrap both http and grpc Stoage.